modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/ConnectionPool.java

```diff
@@ -149,4 +154,41 @@ public void forget(NHttpClientConnection conn) {
             }
         }
     }
+
+    /**
+     * Returns SSL Connections List for the given list host:port combinations
+     *
+     * @param hostList String List in Host:Port format
+     * @return List of NHttpClientConnection
+     */
+    public List<NHttpClientConnection> getSslConnectionsList(Set<String> hostList) {
+        List<NHttpClientConnection> selectedConnections = new ArrayList<NHttpClientConnection>();
+
+        for (String host : hostList) {
+            try {
+                String[] params = host.split(":");
+
+                for (HttpRoute httpRoute : connMap.keySet()) {
+                    if (params.length > 1 && params[0].equalsIgnoreCase(httpRoute.getTargetHost().getHostName())
+                        && (Integer.valueOf(params[1]) == (httpRoute.getTargetHost().getPort())) &&
+                        httpRoute.getTargetHost().getSchemeName().equalsIgnoreCase(sslSchemaName)) {
+
+                        List<NHttpClientConnection> clientConnections = connMap.get(httpRoute);
+
+                        if (clientConnections != null) {
+                            int count = 0;
+                            for (NHttpClientConnection nHttpClientConnection : clientConnections) {
+                                selectedConnections.add(clientConnections.get(count));
+                                count++;
+                            }
+                        }
+                    }
+                }
+            } catch (NumberFormatException e) {
+                log.debug("Error in port number in host:port - " + host + " discard connection loading ", e);
```

Comment:
Why is this a debug message? Shouldn't we inform the user about the wrong config?